src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestRetryFailedManagedIndex.kt

```diff
@@ -0,0 +1,168 @@
+                                    })
+                                }
+                                buildInvalidIndexResponse(builder)
+                                return BytesRestResponse(RestStatus.OK, builder.endObject())
+                            }
+                        }
+                    )
+                } catch (e: ClusterBlockException) {
+                    failedIndices.addAll(listOfIndexMetadata.map {
+                        FailedIndex(it.first.name, it.first.uuid, "failed to update with ClusterBlockException. ${e.message}")
+                    })
+                    buildInvalidIndexResponse(builder)
+                    channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+                }
+            } else {
+                buildInvalidIndexResponse(builder)
+                channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+            }
+        }
+
+        private fun buildInvalidIndexResponse(builder: XContentBuilder) {
+            if (failedIndices.isNotEmpty()) {
+                builder.field(FAILURES, true)
+                builder.startArray(FAILED_INDICES)
+                for (failedIndex in failedIndices) {
+                    builder.startObject()
+                    builder.field("index_name", failedIndex.name)
+                    builder.field("index_uuid", failedIndex.uuid)
+                    builder.field("reason", failedIndex.reason)
+                    builder.endObject()
+                }
+                builder.endArray()
+            } else {
+                builder.field(FAILURES, false)
+            }
+        }
+
+        private fun populateList(state: ClusterState, startState: String?) {
+            for (indexMetadataEntry in state.metaData.indices) {
+                val indexMetadata = indexMetadataEntry.value
+
+                if (indexMetadata.getPolicyName() == null) {
+                    failedIndices.add(FailedIndex(indexMetadata.index.name, indexMetadata.index.uuid, "This index is not being managed."))
+                    continue
+                }
+
+                val managedIndexMetaData = indexMetadata.getManagedIndexMetaData()
+                if (managedIndexMetaData == null) {
+                    failedIndices.add(FailedIndex(indexMetadata.index.name, indexMetadata.index.uuid, "There is no index metadata information"))
+                } else {
+                    if (!isFailedState(managedIndexMetaData)) {
+                        failedIndices.add(FailedIndex(indexMetadata.index.name, indexMetadata.index.uuid, "This index is not in failed state."))
+                    } else {
+                        listOfIndexMetadata.add(Pair(indexMetadata.index,
+                            managedIndexMetaData.copy(failed = false, consumedRetries = 0, transitionTo = startState ?: managedIndexMetaData.transitionTo)))
```

Comment:
We will need it to retry from the first step of an action, so we probably want to reset step information so that the next execution will start over.